### PR TITLE
Move timm vit encoder from bioscan project into modules

### PR DESCRIPTION
**Pull request by amrit110**
# PR Type
Refactor

# Short Description
- Move the ViT transformer from bioscan project into the core lib
- Add get_intermediate_layers function which in turn uses the same API in the ViT model that is wrapped (https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/vision_transformer.py#L780)

# Tests Added
- No tests added

### CODE REVIEW

File: mmlearn/modules/encoders/vision.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        x = self.model.forward_features(x)
+
+        # Separate the class token and patch embeddings
+        cls_token = x[:, 0]
+        patch_embeddings = x[:, 1:]
+
+        return BaseModelOutput(
+            last_hidden_state=patch_embeddings,
+            pooler_output=cls_token,
+            hidden_states=None,
```
Comment (fcogidi) [inline mmlearn/modules/encoders/vision.py:118]: This should return the output of the intermediate layers to match the style of huggingface encoders.

**Review by fcogidi — CHANGES_REQUESTED** [review]

File: mmlearn/modules/encoders/vision.py (hunk 1 above)
Comment (fcogidi) [reply mmlearn/modules/encoders/vision.py:118]: I mean `hidden_states` should not be `None`

Comment (amrit110) [reply mmlearn/modules/encoders/vision.py]: Fixed

**Review by fcogidi — APPROVED** [review]
LGTM

**Pull request merged**